### PR TITLE
[core] Fix per process gpu memory util dashboard stat

### LINKED ISSUE #61456: [core][observability] Ray Dashboard per process GRAM usage measures utilization and not allocated memory
### What happened + What you expected to happen

- Ray dashboard per process GRAM is computed from `pynvml.nvmlDeviceGetProcessesUtilizationInfo().memUtil`. 
- This represents the memory bus utilization (`Percent of time over the past sample period during which global (device) memory was being read or written.` from https://docs.nvidia.com/deploy/nvml-api/structnvmlUtilization__t.html#structnvmlUtilization__t%5B/url%5D), and not how much memory was allocated. 
- As a result, there's a discrepancy between the GRAM usage reported on Ray Dashboard and the `ray_node_gram_used` metric emitted for Prometheus scraping.
https://github.com/ray-project/ray/blob/edefe438250147e6a4012645970a73675614281a/python/ray/dashboard/modules/reporter/gpu_providers.py#L329-L343

### Versions / Dependencies

Ray 2.54 (or latest)

### Reproduction script

```
import os
import time
import ray
import torch

ray.init(address="auto")

@ray.remote(num_gpus=1)
class GpuMemActor:
    def __init__(self, gib=8):
        self.gib = gib
        self.physical_gpu_index = ray.get_runtime_context().get_accelerator_ids()["GPU"][0]
        n = int(gib * (1024**3) / 4)
        self.buf = torch.empty(n, dtype=torch.float32, device="cuda")
        self.buf.fill_(1.0)

    def info(self):
        return {
            "pid": os.getpid(),
            "host": os.uname().nodename,
            "gpu_index": self.physical_gpu_index,
            "allocated_mib": torch.cuda.memory_allocated() / (1024**2),
            "reserved_mib": torch.cuda.memory_reserved() / (1024**2),
        }

    def burn(self, seconds=1200):
        end = time.time() + seconds
        while time.time() < end:
            self.buf.mul_(1.000001)
            torch.cuda.synchronize()
        return True

a = GpuMemActor.options(name="gpu-mem-repro").remote(gib=8) 
print("Actor info:", ray.get(a.info.remote()))
a.burn.remote(1200)

while True:
    print(ray.get(a.info.remote()))
    time.sleep(30)
```

Image

Image

### Issue Severity

P1

### PR DESCRIPTION
**Pull request by dayshah**
## Description
The way we got the per process used gpu memory that's displayed on the actor dashboard was wrong.
image

We would use memutil from nvmlDeviceGetProcessesUtilizationInfo and then divide by 100 and multiply by the total memory on the gpu. But memutil is actually the percentage of time the memory bus is busy, not the percentage of memory used by the process...

Someone else ran into this same misunderstanding in the past - https://forums.developer.nvidia.com/t/memory-metric-reported-by-nvml-and-nvdia-smi-seems-to-differ/75282
Nvidia docs (https://docs.nvidia.com/deploy/nvml-api/structnvmlUtilization__t.html#structnvmlUtilization__t%5B/url%5D) define memutil as 
```
Percent of time over the past sample period during which global (device) memory was being read or written.
```

The fallback actually got the used memory for the process correctly, so now I'm just using that to get the used memory while I use the newer api to just get sm utilization.

Fixes https://github.com/ray-project/ray/issues/61456

### CODE REVIEW

**Comment by edoakes** [comment]
testable?

File: python/ray/tests/test_gpu_provider_on_gpu.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            gpu_info = single_gpu_info
+            break
+
+    assert process_info is not None
+
+    reported_proc_mb = process_info["gpu_memory_usage"]
+    # Proc memory usage should be at least the tensor size and within 100mb
+    assert (
+        reported_proc_mb >= tensor_size_mb and reported_proc_mb - tensor_size_mb < 100
+    )
```
[the remark this answers is not in the document]
Comment (dayshah) [reply python/ray/tests/test_gpu_provider_on_gpu.py]: true

**Comment by dayshah** [comment]
> testable?

added test on gpu

**Review by edoakes — APPROVED** [review]

File: python/ray/dashboard/modules/reporter/gpu_providers.py [hunk 2]
```diff
@@ -318,56 +318,63 @@ def _get_gpu_info(self, gpu_handle, gpu_index: int) -> Optional[GpuUtilizationIn
                 )
                 utilization = int(utilization_info.gpu)
             except self._pynvml.NVMLError as e:
-                logger.debug(f"Failed to retrieve GPU utilization: {e}")
+                logger.debug(
+                    f"Failed to retrieve GPU utilization via `nvmlDeviceGetUtilizationRates`: {e}"
+                )
```
Comment (edoakes) [inline python/ray/dashboard/modules/reporter/gpu_providers.py]: seems like it should be higher level than debug?

Comment (dayshah) [reply python/ray/dashboard/modules/reporter/gpu_providers.py]: I think it can be driver dependent, i'm a little concerned it can start spamming ppl who don't have the perfect setup, put it up to info for now

But keeping the ones that require a relatively newer driver version at the bottom at debug, so it doesn't spam

Comment (dayshah) [reply python/ray/dashboard/modules/reporter/gpu_providers.py]: Actually just doing log once at info level for all so a user can figure out something's up but it doesn't spam

**Review by Sparks0219 — APPROVED** [review]
🚢

**Comment by dayshah** [comment]
> 🚢 

does the test work?

**Comment by Sparks0219** [comment]
> does the test work?

RIP a bit too well, it looks like it passes before and after your PR. The reported_proc_mb didn't change with the different nvml API, stayed at 2164 MB
